src/libraries/Microsoft.PowerFx.Core/Public/Values/NumberValue.cs

```diff
@@ -22,7 +22,7 @@ public override void Visit(IValueVisitor visitor)
 
         public override void ToExpression(StringBuilder sb, FormulaValueSerializerSettings settings)
         {
-            sb.Append(Value.ToString());
+            sb.Append((Value == 0) ? "0" : Value.ToString());
```

Comment:
>(Value == 0) ? "0" : Value.ToString() [](http://example.com/codeflow?start=22&length=37)

We were getting a "-0" result in a few scenarios of a very small value. This fixes it.